apiserver/facades/controller/instancepoller/merge.go

```diff
@@ -67,19 +75,70 @@ func (o *mergeMachineLinkLayerOp) Build(_ int) ([]txn.Op, error) {
 	return ops, nil
 }
 
+// normaliseIncoming is intended accommodate providers such as EC2
+// that know device hardware addresses, but not device names.
+// We populate names on the incoming data based on
+// matching existing devices by hardware address.
+// If we locate multiple existing devices with the hardware address,
+// such as will be the case for bridged NICs, fallback through the
+// following options.
+// - If there is a device that already has a provider ID, use that name.
+// - If the devices are of different types, choose an ethernet device over
+//   a bridge (as observed for MAAS).
+func (o *mergeMachineLinkLayerOp) normaliseIncoming() {
+	incoming := o.Incoming()
+
+	// If the incoming devices have names, no action is required
+	// (assuming all or none here).
+	if len(incoming) > 0 && incoming[0].InterfaceName != "" {
```

Comment:
Is this a bit optimistic, can we safely assume that because `0` has an `InterfaceName` everyone will (I don't know, genuine question)?

Reply:
Based on current implementations, I believe so. Added a comment to this effect.